emily_cron/app/services/deposit_processor.py

```diff
@@ -160,6 +214,49 @@ def update_deposits(self) -> None:
 
         logger.info("Deposit status update job completed")
 
+    def _group_rbf_transactions(self, rbf_txs: list[EnrichedDepositInfo]) -> dict[str, set[str]]:
+        """Group RBF transactions by their replacement chains.
+        Args:
+            rbf_txs: List of transactions with RBF replacements
+        Returns:
+            dict[str, set[str]]: Dictionary mapping group IDs to sets of transaction IDs
+        """
+        rbf_groups: dict[str, set[str]] = {}
+
+        # First, build groups of related transactions (original + replacements)
+        for tx in rbf_txs:
+            # Create a set of all txids in this RBF chain
+            chain_txids = set(tx.rbf_txids)
+            chain_txids.add(tx.bitcoin_txid)
+
+            # Find all groups that overlap with this chain
+            overlapping_groups = []
+            for group_id, group_txids in list(rbf_groups.items()):
```

Comment:
I think the `list` is unnecessary here, since `items()` should return a generator.